velox/expression/SignatureBinder.h

```diff
@@ -43,13 +43,22 @@ class SignatureBinder {
   // Returns true if successfully resolved all generic type names.
   bool tryBind();
 
+  // Return true if actualType can bind to typeSignature and update bindings_.
+  bool tryBind(
```

Comment:
To avoid exposing implementation details, consider introducing a base class SignatureBinderBase with this method as a protected method. Then, define SignatureBinder and ReverseSignatureBinder using SignatureBinderBase as the base class.

I'd also move ReverseSignatureBinder class into this file or into ReverseSignatureBinder.h file in this directory.